client/asset/dcr/dcr.go

```diff
@@ -1521,19 +1522,13 @@ func (dcr *ExchangeWallet) AuditContract(coinID, contract, txData dex.Bytes, reb
 	// Validate contract output.
 	// Script must be P2SH, with 1 address and 1 required signature.
 	contractTxOut := contractTx.TxOut[vout]
-	scriptClass, addrs, sigsReq, err := txscript.ExtractPkScriptAddrs(contractTxOut.Version, contractTxOut.PkScript, dcr.chainParams, false)
-	if err != nil {
-		return nil, fmt.Errorf("error extracting script addresses from '%x': %w", contractTxOut.PkScript, err)
-	}
-	if scriptClass != txscript.ScriptHashTy {
+	scriptClass, addrs := stdscript.ExtractAddrs(contractTxOut.Version, contractTxOut.PkScript, dcr.chainParams)
+	if scriptClass != stdscript.STScriptHash {
 		return nil, fmt.Errorf("unexpected script class %d", scriptClass)
 	}
 	if len(addrs) != 1 {
 		return nil, fmt.Errorf("unexpected number of addresses for P2SH script: %d", len(addrs))
 	}
-	if sigsReq != 1 {
```

Comment:
Thanks for calling this out.  I believe this was a "why not" kind of check since in this method (`AuditContract`), we ensure the script class is `STScriptHash` with just a single p2sh address successfully decoded from the pkScript (noting that as per our [recent discussion](https://github.com/decred/dcrd/pull/2807#discussion_r752492737) `ExtractAddrs` may return a standard script class even if address decoding failed).

In version 0 scripts, stdscript claims 1 sig for `STScriptHash`, but how meaningful that is is not clear (the number of signatures actually required to spend depends on the SH script).  The important script check other than the pkScript class and address checks is the contract sh script done separately.

However, I don't mind putting back the `sigsReq` check with `stdscript.DetermineRequiredSigs` if my reasoning is off.

Reply:
Personally, I think it's more than fine to remove the check, because, for the most part, the number of required signatures is mostly legacy baggage brought along for the ride and, realistically, has dubious meaning in various circumstances.

As you pointed out, once such example is for `STScriptHash`, which is just an assumption that there will be at least one signature required, but you really need the redeem script to determine anything more.  However, it's even more ambiguous when you start talking about threshold signatures which just appear on-chain to require a single signatures, but in reality, it might actually need hundreds.

I mainly just wanted to call out that it was being removed so there was no chance of it looking like an accident.